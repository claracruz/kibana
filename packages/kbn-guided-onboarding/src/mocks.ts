/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { GuideCardSolutions, GuideCardsProps } from './components/landing_page/guide_cards';
import { GuideCardConstants } from './components/landing_page/guide_cards.constants';
import { GuideId, GuideStatus, GuideStepIds, StepStatus } from '..';

const guideId: GuideId = 'testGuide';
const status: GuideStatus = 'not_started';
const stepStatus: StepStatus = 'inactive';
const testGuideId: GuideStepIds = 'step1';

export const activateGuideMock: GuideCardsProps & { card: GuideCardConstants } = {
  activateGuideDefaultState: jest.fn(),
  navigateToApp: jest.fn(),
  activeFilter: 'all',
  guidesState: [
    {
      guideId,
      status,
      isActive: false,
      steps: [
        {
          id: testGuideId,
          status: stepStatus,
        },
      ],
    },
  ],
  card: {
    solution: 'search' as GuideCardSolutions,
    title: 'test_title',
    guideId: 'testGuide',
    telemetryId: 'test_telemetry_id',
    order: 1,
  },
};

export const navigateToAppGuideMock: GuideCardsProps & { card: GuideCardConstants } = {
  ...activateGuideMock,
  card: {
    solution: 'search' as GuideCardSolutions,
    title: 'test_title',
    telemetryId: 'test_telemetry_id',
    order: 1,
    navigateTo: {
      appId: 'test_appId',
      path: 'path/to/app',
    },
  },
};
