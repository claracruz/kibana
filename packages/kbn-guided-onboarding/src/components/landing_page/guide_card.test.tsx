/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { mount, shallow } from 'enzyme';

import { GuideCard } from './guide_card';
import { EuiCard } from '@elastic/eui';
import { act } from 'react-dom/test-utils';
import { activateGuideMock, navigateToAppGuideMock } from '../../mocks';

describe('guide cards', () => {
  describe('snapshots', () => {
    test('should render card', async () => {
      const component = await shallow(<GuideCard {...activateGuideMock} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Navigation', () => {
    test('Should activate guide default state if has guide ID', async () => {
      const props = {
        ...activateGuideMock,
        activateGuideDefaultState: jest.fn(),
      };
      const component = await mount(<GuideCard {...props} />);
      const { activateGuideDefaultState, guidesState } = props;
      const card = component.find(EuiCard);

      await act(async () => {
        // @ts-ignore
        card.props().onClick({});
      });
      component.update();

      expect(activateGuideDefaultState).toHaveBeenCalledTimes(1);
      expect(activateGuideDefaultState).toBeCalledWith(guidesState[0].guideId, guidesState[0]);
    });

    test('Should navigate to guide path if navigation path provided', async () => {
      const props = {
        ...navigateToAppGuideMock,
        activateGuideDefaultState: jest.fn(),
      };
      const component = await mount(<GuideCard {...props} />);
      const card = component.find(EuiCard);

      await act(async () => {
        // @ts-ignore
        card.props().onClick({});
      });
      component.update();

      const { activateGuideDefaultState } = props;
      const {
        navigateToApp,
        card: { navigateTo },
      } = navigateToAppGuideMock;
      expect(activateGuideDefaultState).toHaveBeenCalledTimes(0);
      expect(navigateToApp).toHaveBeenCalledTimes(1);
      expect(navigateToApp).toBeCalledWith(navigateTo?.appId, { path: navigateTo?.path });
    });
  });
});
