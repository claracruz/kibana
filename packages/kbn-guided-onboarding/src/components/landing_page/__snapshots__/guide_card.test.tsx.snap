// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`guide cards snapshots should render card 1`] = `
<EuiCard
  betaBadgeProps={
    Object {
      "label": "search",
    }
  }
  css={
    Object {
      "map": undefined,
      "name": "9w6uxv",
      "next": undefined,
      "styles": "
  position: relative;
  min-height: 110px;
  width: 380px;
  .euiCard__content {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
",
      "toString": [Function],
    }
  }
  data-test-subj="test_telemetry_id"
  description={<React.Fragment />}
  hasBorder={false}
  isDisabled={false}
  onClick={[Function]}
  title={
    <React.Fragment>
      <EuiSpacer
        size="s"
      />
      <h3
        style={
          Object {
            "fontWeight": 600,
          }
        }
      >
        test_title
      </h3>
    </React.Fragment>
  }
  titleSize="xs"
/>
`;
